servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/HttpClients.java

```diff
@@ -36,27 +38,59 @@
 import io.servicetalk.transport.api.HostAndPort;
 import io.servicetalk.transport.netty.internal.BuilderUtils;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import java.net.InetSocketAddress;
 import java.net.SocketAddress;
 import java.util.Collection;
+import java.util.List;
 import java.util.function.Function;
 
 import static io.servicetalk.concurrent.api.AsyncCloseables.emptyAsyncCloseable;
 import static io.servicetalk.concurrent.api.Publisher.failed;
 import static io.servicetalk.http.netty.GlobalDnsServiceDiscoverer.globalDnsServiceDiscoverer;
 import static io.servicetalk.http.netty.GlobalDnsServiceDiscoverer.globalSrvDnsServiceDiscoverer;
 import static io.servicetalk.http.netty.GlobalDnsServiceDiscoverer.mappingServiceDiscoverer;
+import static io.servicetalk.utils.internal.ServiceLoaderUtils.loadProviders;
 import static java.util.function.Function.identity;
 
 /**
  * Factory methods for building {@link HttpClient} (and other API variations) instances.
  */
```

Comment:
You may want to references `HttpProviders` or the client provider interfaces here. "You can customize the creation of clients using the `SingleAddressHttpClientBuilderProvider` interface".